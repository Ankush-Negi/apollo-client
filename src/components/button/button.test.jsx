import React from 'react';
import { mount } from 'enzyme';
import Button from './button';

const props = {
  value: 'Submit',
};

describe('Button Component', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<Button {...props} />);
  });
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('render button click', () => {
    const clicked = wrapper.find(Button).simulate('click');
    expect(clicked.length).toBe(1);
  });
});
