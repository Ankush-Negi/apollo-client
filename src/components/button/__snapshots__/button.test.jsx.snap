// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render correctly 1`] = `
<Button
  color="primary"
  disabled={false}
  style={Object {}}
  value="Submit"
>
  <styled.button
    color="primary"
    disabled={false}
    id="Button1"
    style={Object {}}
    value="Submit"
  >
    <button
      className="sc-AxjAm gWioyM"
      color="primary"
      disabled={false}
      id="Button1"
      style={Object {}}
      value="Submit"
    >
      <b>
        Submit
      </b>
    </button>
  </styled.button>
</Button>
`;
